src/datamodel/serializer/Uuid.cpp

```diff
@@ -10,3 +10,122 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/datamodel/serializer/Uuid.h"
+
+#include <arpa/inet.h>
+
+#include <iomanip>
+#include <sstream>
+#include <stdexcept>
+
+namespace uprotocol::datamodel::serializer::uuid {
+
+std::string AsString::serialize(const uprotocol::v1::UUID uuid) {
+	// Extracting the parts of the UUIDv8
+	uint64_t unix_ts_ms = (uuid.msb() >> 16) & 0xFFFFFFFFFFFF;  // 48 bit
+	uint8_t ver = (uuid.msb() >> 12) & 0xF;                     // 4 bit
+	uint16_t counter = uuid.msb() & 0xFFF;                      // 12 bit
+	uint8_t var = (uuid.lsb() >> 62) & 0x3;                     // 2 bit
+	uint64_t rand_b = uuid.lsb() & 0x3FFFFFFFFFFFFFFF;          // 62 bit
+
+	// Formatting the UUIDv8 in the traditional format
+	std::stringstream ss;
+	ss << std::hex << std::setfill('0') << std::setw(8)
+	   << ((unix_ts_ms >> 16) & 0xFFFFFFFF)  // First 32 bits of timestamp
+	   << "-" << std::setw(4)
+	   << ((unix_ts_ms)&0xFFFF)  // Next 16 bits of timestamp i.e. last 16 bit
+	                             // of ts
+	   << "-" << std::setw(4)
+	   << (((ver & 0xF) << 12) |
+	       (counter & 0xFFF))  // Last 16 bits of timestamp and version
+	   << "-" << std::setw(4)
+	   << (((var & 0x3) << 14) |
+	       ((rand_b >> 48) & 0x3FFF))  // Variant and randb
+	   << "-" << std::setw(12) << (rand_b & 0xFFFFFFFFFFFF);  // Random number
+
+	return std::move(ss).str();
+}
+
+uprotocol::v1::UUID AsString::deserialize(const std::string& str) {
+	// Check if the UUID string is in the correct format
```

Comment:
NITPICK: How about a comment showing the "correct format"

// format : 12345678-1234-1234-1234-1234567890123
// index  : 012345678901234567890123456789012345

That would look better in monospace, but I can't figure out how to do it in a comment.